portal-impl/src/com/liferay/portal/util/PortalInstances.java

```diff
@@ -66,17 +68,7 @@
 public class PortalInstances {
 
 	public static void addCompanyId(long companyId) {
-		if (ArrayUtil.contains(_companyIds, companyId)) {
-			return;
-		}
-
-		long[] companyIds = new long[_companyIds.length + 1];
-
-		System.arraycopy(_companyIds, 0, companyIds, 0, _companyIds.length);
-
-		companyIds[_companyIds.length] = companyId;
-
-		_companyIds = companyIds;
+		_companyIds.addLast(companyId);
```

Comment:
@dantewang Based on this checking "ArrayUtil.contains(_companyIds, companyId)", the elements in the collection should be unique, but Deque can not make sure the entries are unique, let's use ConcurrentSkipListSet.